include/psa/crypto_types.h

```diff
@@ -90,7 +90,15 @@ typedef uint32_t psa_key_lifetime_t;
 
 /** Encoding of identifiers of persistent keys.
  */
+/* Mbed Crypto particularity: in an SPM integration, psa_key_id_t is a larger
+ * type that encodes both the application-supplied key identifier and the
+ * identity of the owner of the key. This non-compliant implementation is only
+ * used when building Mbed Crypto as a library inside the PSA Cryptography
+ * service. Applications that call the service see the standard API with a
+ * 32-bit type. */
+#if !defined(MBEDTLS_PSA_CRYPTO_SPM)
```

Comment:
`MBEDTLS_PSA_CRYPTO_SPM` is defined for the entire secure side. This means that if another _secure_ partition will try to communicate with crypto, this client will see `psa_key_id_t` as the full struct (i.e. `psa_key_file_id_t`) as not as `uint32_t`.

Reply:
This could work if the ifdefs in the header files are tested for `PSA_CRYPTO_SECURE` instead of `MBEDTLS_PSA_CRYPTO_SPM`.

Reply:
Ok, I'll change that.